File: 2nd Year/2nd Semester/MPP/PROIECT-JAVA/Laborator/TriatlonModel/src/main/java/triatlon/model/Rezultat.java
package triatlon.model;

import java.util.Objects;

public class Rezultat extends Entity<Integer> {
    private Participant participant;
    private Arbitru arbitru;
    private TipProba tipProba;
    private int punctaj;

    public Rezultat(Integer integer, Participant participant, Arbitru arbitru, TipProba tipProba, int punctaj) {
        super(integer);
        this.participant = participant;
        this.arbitru = arbitru;
        this.tipProba = tipProba;
        this.punctaj = punctaj;
    }

    public Rezultat(Integer integer, Participant participant, Arbitru arbitru, String tipProba, int punctaj) {
        super(integer);
        this.participant = participant;
        this.arbitru = arbitru;
        this.tipProba = TipProba.valueOf(tipProba);
        this.punctaj = punctaj;
    }

    public Participant getParticipant() {
        return participant;
    }

    public void setParticipant(Participant participant) {
        this.participant = participant;
    }

    public Arbitru getArbitru() {
        return arbitru;
    }

    public void setArbitru(Arbitru arbitru) {
        this.arbitru = arbitru;
    }

    public TipProba getTipProba() {
        return tipProba;
    }

    public String getTipProbaString(){
        return tipProba.toString();
    }

    public void setTipProba(TipProba tipProba) {
        this.tipProba = tipProba;
    }

    public int getPunctaj() {
        return punctaj;
    }

    public void setPunctaj(int punctaj) {
        this.punctaj = punctaj;
    }

    @Override
    public String toString() {
        return "Rezultat{" +
                "participant=" + participant +
                ", tipProba=" + tipProba +
                ", punctaj=" + punctaj +
                ", arbitru=" + arbitru +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (o == null || getClass() != o.getClass()) return false;
        Rezultat rezultat = (Rezultat) o;
        return getPunctaj() == rezultat.getPunctaj() &&
                Objects.equals(getParticipant(), rezultat.getParticipant()) &&
                tipProba == rezultat.tipProba &&
                Objects.equals(getArbitru(), rezultat.getArbitru());
    }

    @Override
    public int hashCode() {
        return Objects.hash(getParticipant(), tipProba, getPunctaj(), getArbitru());
    }
}

File: 2nd Year/2nd Semester/MPP/PROIECT-C#/Lab1/Lab1/Domain/Rezultat.cs
namespace Lab1.Domain {
    public class Rezultat : Entity<int> {
        public Participant Participant { get; set; }
        public Arbitru Arbitru { get; set; }
        public TipProba TipProba { get; set; }
        public int Points { get; set; }

        public Rezultat(int id, Participant participant, Arbitru arbitru, TipProba proba, int points) {
            this.Id = id;
            this.Participant = participant;
            this.Arbitru = arbitru;
            this.TipProba = proba;
            this.Points = points;
        }

        public override string ToString() {
            return this.Participant + " " + this.Arbitru + " " + this.TipProba + " " + this.Points;
        }

        public override bool Equals(object obj) {
            if (this == obj) {
                return true;
            }
            if (obj == null || this.GetType() != obj.GetType()) {
                return false;
            }
            Rezultat rezultat = (Rezultat)obj;
            return this.Id == rezultat.Id;
        }

        public override int GetHashCode() {
            return this.Id;
        }
    }
}
